src/commands.ts

```diff
@@ -383,36 +387,39 @@ async function openWorkspace(
     newWindow = false
   }
 
-  // If a folder isn't specified, we can try to open a recently opened folder.
-  if (!folderPath) {
+  // If a folder isn't specified or we have been asked to open the most recent,
+  // we can try to open a recently opened folder/workspace.
+  if (!folderPath || openRecent) {
     const output: {
       workspaces: { folderUri: vscode.Uri; remoteAuthority: string }[]
     } = await vscode.commands.executeCommand("_workbench.getRecentlyOpened")
     const opened = output.workspaces.filter(
-      // Filter out `/` since that's added below.
+      // Remove recents that do not belong to this connection.  The remote
+      // authority maps to a workspace or workspace/agent combination (using the
+      // SSH host name).  This means, at the moment, you can have a different
+      // set of recents for a workspace versus workspace/agent combination, even
+      // if that agent is the default for the workspace.
```

Comment:
Do we have any docs on how this works? The comment seems well-written, but I feel like I'm still not quite getting it because of missing context

Reply:
Nope, not that I know of.  I am honestly not really sure how to even write the comment, I rewrote it several times and still feel like I am not really explaining it well.

The problem is that both of these go to the same place:

```
ssh vscode-coder--dev27.dev
ssh vscode-coder--dev27 # opens dev agent because dev is the default
```

But VS Code has no idea these are the same, they are different strings after all.  So depending on which you use you could get a different set of recents.

When clicking from the dashboard we seem to use the second form (without the agent name).

I think probably what we should do is always append the agent name if it is missing, and never use the workspace-only form.  Then this will not be a problem at all.  But for now I just leave the comment.

Reply:
And to further clarify, this only arises when using the sidebar in VS Code, where you can both connect to a workspace or directly to an agent.